modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +608,200 @@ private void deleteApplication(String applicationId) throws Exception {
+        List<Map<String, String>> expectedLastOrganizations =
+                createdOrganizations.subList(totalCreatedOrganizations - limit, totalCreatedOrganizations);
+        Collections.reverse(expectedLastOrganizations); // Reverse the order to match the expected response
+        Assert.assertEquals(actualOrganizationsSimplified, expectedLastOrganizations,
+                "The previous page should return the last created organizations.");
+
+
+        // Step 4: Test with orgLimit (equal to total org count)
+        validatePaginationForNonPaginatedLimit(orgLimit);
+
+        // Step 5: Test with largeLimit (greater than total org count)
+        validatePaginationForNonPaginatedLimit(largeLimit);
+    }
+
+    private int createOrganizations(int numberOfOrganizations) throws JSONException {
+
+        createdOrganizations = new ArrayList<>();
+        int createdOrganizationsCount = 0;
+
+        for (int i = 0; i < numberOfOrganizations; i++) {
+            JSONObject body = new JSONObject()
+                    .put(ORGANIZATION_NAME, String.format("Org-%d", i))
+                    .put(ORGANIZATION_DESCRIPTION, String.format("Test Organization %d", i));
+
+            Response response =
+                    getResponseOfPostWithOAuth2(ORGANIZATION_MANAGEMENT_API_BASE_PATH, body.toString(), m2mToken);
+
+            if (response.getStatusCode() == HttpStatus.SC_CREATED) {
+                // Store the created organization details
+                Map<String, String> org = new HashMap<>();
+                org.put(ORGANIZATION_NAME, String.format("Org-%d", i));
+                org.put(ORGANIZATION_DESCRIPTION, String.format("Test Org-%d", i));
+                createdOrganizations.add(org);
+
+                createdOrganizationsCount++;
+            } else {
+                throw new RuntimeException("Failed to create organization " + i);
+            }
+        }
+
+        return createdOrganizationsCount;
+    }
+
+    private void validatePaginationForNonPaginatedLimit(int limit) {
+
+        String limitUrl = ORGANIZATION_MANAGEMENT_API_BASE_PATH + LIMIT_ARG + limit + "&recursive=false";
+        Response response = getResponseOfGetWithOAuth2(limitUrl, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        List<Map<String, String>> links = response.jsonPath().getList("links");
+
+        Assert.assertNull(links, "Links should be null when all organizations are returned in one page.");
+
+        int returnedOrgCount = response.jsonPath().getList("organizations").size();
+        Assert.assertEquals(returnedOrgCount, totalCreatedOrganizations,
+                "The number of returned organizations should match the total created organizations.");
+    }
+
+    private String getLink(List<Map<String, String>> links, String rel) {
+
+        for (Map<String, String> link : links) {
+            if (rel.equals(link.get("rel"))) {
+                String href = link.get("href");
```

Comment:
Let's extract "rel" and "href" to constants as well.

Reply:
Done